Flaky: test_http_endpoint_missing_auth_value

## Flaky Test

* **Test Name:** test_http_endpoint_missing_auth_value
* **Link:** [test_http_endpoint.py#L181.](https://github.com/elastic/beats/blob/45cff00533f891a61c9fb95ea9e76b0c8a374585/x-pack/filebeat/tests/system/test_http_endpoint.py#L181)
* **Branch:** master
* **Notes:** spotted in [first](https://github.com/elastic/beats/runs/4949388049?check_suite_focus=true) [second](https://github.com/elastic/beats/runs/4949253293?check_suite_focus=true)

### Stack Trace

```
=================================== FAILURES ===================================
__________________ Test.test_http_endpoint_missing_auth_value __________________

self = <test_http_endpoint.Test testMethod=test_http_endpoint_missing_auth_value>

        def test_http_endpoint_missing_auth_value(self):
            """
            Test http_endpoint input with missing basic auth values.
            """
            options = """
      basic_auth: true
      username: testuser
      password:
    """
            self.get_config(options)
            filebeat = self.start_beat()
>           self.wait_until(lambda: self.log_contains(
Error: running "/Users/runner/work/beats/beats/build/ve/darwin/bin/pytest --timeout=90 --durations=20 --junit-xml=build/TEST-python-unit.xml tests/system/test_filebeat_xpack.py tests/system/test_http_endpoint.py tests/system/test_xpack_modules.py" failed with exit code 1
                "username and password required when basicauth is enabled"))

tests/system/test_http_endpoint.py:192: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_http_endpoint.Test testMethod=test_http_endpoint_missing_auth_value>
cond = <function Test.test_http_endpoint_missing_auth_value.<locals>.<lambda> at 0x10c55fa60>
max_timeout = 10, poll_interval = 0.1, name = 'cond'

    def wait_until(self, cond, max_timeout=10, poll_interval=0.1, name="cond"):
        """
        Waits until the cond function returns true,
        or until the max_timeout is reached. Calls the cond
        function every poll_interval seconds.
    
        If the max_timeout is reached before cond() returns
        true, an exception is raised.
        """
        start = datetime.now()
        while not cond():
            if datetime.now() - start > timedelta(seconds=max_timeout):
>               raise TimeoutError("Timeout waiting for '{}' to be true. ".format(name) +
                                   "Waited {} seconds.".format(max_timeout))
E               beat.beat.TimeoutError: Timeout waiting for 'cond' to be true. Waited 10 seconds.
```

--- a/x-pack/filebeat/tests/system/test_http_endpoint.py
+++ b/x-pack/filebeat/tests/system/test_http_endpoint.py
@@ -246,15 +246,20 @@ class Test(BaseTest):
         assert r.json()[
             'message'] == 'wrong Content-Type header, expecting application/json'
 
-    @unittest.skipIf(os.getenv("CI") is not None and platform.system() == 'Darwin', 'Flaky test: https://github.com/elastic/beats/issues/30028')
     def test_http_endpoint_missing_auth_value(self):
         """
         Test http_endpoint input with missing basic auth values.
         """
+        # Include a benchmark input to keep the beat alive, otherwise it might
+        # shut down before logging the error we're looking for
         options = """
   basic_auth: true
   username: testuser
   password:
+#- type: benchmark
+#  enabled: true
+#  message: "placeholder"
+#  eps: 1
 """
         self.get_config(options)
         filebeat = self.start_beat()

--- a/x-pack/filebeat/tests/system/test_http_endpoint.py
+++ b/x-pack/filebeat/tests/system/test_http_endpoint.py
@@ -256,10 +256,10 @@ class Test(BaseTest):
   basic_auth: true
   username: testuser
   password:
-#- type: benchmark
-#  enabled: true
-#  message: "placeholder"
-#  eps: 1
+- type: benchmark
+  enabled: true
+  message: "placeholder"
+  eps: 1
 """
         self.get_config(options)
         filebeat = self.start_beat()

